### DISCUSSION TITLE
How come WorldWindKotlin license is Apache 2.0 and WorldWind License is NASA license

### DISCUSSION THREAD

**Discussion by onderaltintas** [WorldWindEarth/WorldWindKotlin#72] [Q&A] [answered]
Hi

When I inspected NASA license, it didn't look like permissive enough to outside of the USA. However WorldWindKotlin is Apache 2.0 licensed that I can use. If WWK is using WW sources, how come it is licensed under Apache 2.0 or can I use the wwk source freely under the restrictions of Apache 2.0? That makes a bit confusion. 

Thank you for the answers,
Have a nice life!

**Comment by EMaksymenko** [comment, marked as answer]
WWK does not use nether Java nor JS sources of WorldWind. It was completely rewritten from skratch on Kotlin language using only architecture concept from different codebases of WorldWind, which makes it completely independent, but WWW and WWJ are under Apache license, thats why we keep the same approach. Only WWA has a NASA license.
